### ISSUE TITLE
Allow user to use Lookup within Transactions for existing Organization Contacts-Part 1

### ISSUE THREAD

**Issue by AureliaKhorsand** [fecgov/fecfile-web-app#548]
Business Reason

The system will employ a Lookup feature when entering transactions in order to allow the user to pull existing contact information from the committee account's Contact Management table. This will also assist the FEC in making more accurate data available to the public.

Acceptance Criteria

_Note: The transaction entry form will _only_ show the contact lookup until a contact is selected. Once a contact is selected, the transaction form will show._

***Given*** the user is entering a transaction

  ***And*** they have indicated an Entity Type of "ORG", (or if "ORG" is the required entity type)

  ***Then*** they will be able to search their committee account's Contact Management table by _name_ using a Lookup feature to find and select an existing contact

  ***And*** the results will including the following information in the dropdown under the text "Select an existing organization contact":

*   Organization Name
*   "In Contacts"

***And*** the Lookup results will:

*   highlight the matching text from the search box within the results
*   highlight a selection grey if hovered over with the pointer
*   be limited to 10 maximum _(Dev/QA note: for now, users need to type more to narrow it down. #542 will address when results are more than 10.)_

***And*** when I select the contact I want, the system will pull in the contact information necessary for the transaction and show the full transaction entry form.

  ***If*** there are no results, the system will indicate as such.

BA notes

FECFile Online basic requirements as follows:

*   Users will be able to search their committee account's Contact Management table using a Lookup feature to find and select an existing contact for a transaction
*   When the user selects the contact, the system will pull in the contact information necessary for the transaction.
*   The transaction will be associated with that contact (used for aggregations/itemization logic)

DEV notes

From this point on, contacts and transactions need to be connected. _Each transaction must be associated with a contact, such as a UUID._

QA notes

Relevant existing Transactions to test:

*   Tribal Receipt
*   Offsets to Operating Expenditures
*   Other Receipts

To test, identify an existing organization contact in Contact Management for your committee account and use that name to search.

NOTE: Other Receipts is Group C therefore Committee ID field is not displayed.

Wireframes: 

  (Note: down arrow next to "Organization" only if other entity types are valid for selection.)

![image.png]([https://images.zenhubusercontent.com/61a50e5fdcda4774a0cf2a0e/4c94ddb8-a8a9-47ab-8ba1-f6cfe8c0a815](https://images.zenhubusercontent.com/61a50e5fdcda4774a0cf2a0e/4c94ddb8-a8a9-47ab-8ba1-f6cfe8c0a815))

![image.png]([https://images.zenhubusercontent.com/61a50e5fdcda4774a0cf2a0e/9cb37d15-a946-464a-aa07-7b56b4550852](https://images.zenhubusercontent.com/61a50e5fdcda4774a0cf2a0e/9cb37d15-a946-464a-aa07-7b56b4550852))

![image.png]([https://images.zenhubusercontent.com/61a50e5fdcda4774a0cf2a0e/6c6a3601-895d-4ac9-bb9c-681d09c52a87](https://images.zenhubusercontent.com/61a50e5fdcda4774a0cf2a0e/6c6a3601-895d-4ac9-bb9c-681d09c52a87))

### QA Notes ###
null

### DEV Notes ###
null

### Design ###
null

 See full ticket and images here: [FECFILE-1024](https://fecgov.atlassian.net/browse/FECFILE-1024)

**Comment by WiseQA** [comment]
QA review  of ticket #72 impacts this ticket also.   "Address"  fields not updating when using the lookup and selecting a Contact.

@dheitzer @AureliaKhorsand @GreggMoreland per meeting returning to In-Progress for fix.

**Comment by dheitzer** [comment]
[#538](https://app.zenhub.com/workspaces/fecfile-online-619e578e68408b001c831251/issues/fecgov/fecfile-web-app/538) is built on this ticket and can be reviewed once it is merged into dev.

**Comment by mjtravers** [comment]
Sending ticket from CR to In Progress to address @toddlees  change requests.

**Comment by dheitzer** [comment]
CR changes pushed.

**Comment by mjtravers** [comment]
Passes CR, sending to QA

**Comment by WiseQA** [comment]
QA review verified lookup is searchable and contains Organization name and verified In Contacts.  (Note also verified that the highlight a selection grey if hovered over with the pointer.)

![image.png](https://images.zenhubusercontent.com/61ba01e428a658b4eb0ca758/db359ee0-046a-4142-9a84-c4fb0e3ee8d4)

Once contact selected verified Contributor fields and Address fields updated with contact information.

![image.png](https://images.zenhubusercontent.com/61ba01e428a658b4eb0ca758/7a2ff66b-c1c1-4ac5-b0a2-b9f978093827)

During testing of the Tribal Receipt, Offsets to Operating Expenditures,  and Other Receipts transactions once select "Save and add another" button, a blank receipt is displayed but the "Searchable" field is not clearing out.

![image.png](https://images.zenhubusercontent.com/61ba01e428a658b4eb0ca758/ebb64da0-95dc-4d4f-9aad-8e3736f0d59e)

@dheitzer @MitchellTCG returned to In-Progress for fix 
(Note Ticket #72 will need updated also per our discussion)

**Comment by mjtravers** [comment]
Lookup box is now cleared after the user selects a contact and copies it into the the transaction input fields. The lookup box was also added to the JF TRAN screen. Passed CR so sending to QA.

**Comment by WiseQA** [comment]
QA review and verified "Search" field is cleared out once "Save and add another" button is selected.

![image.png](https://images.zenhubusercontent.com/61ba01e428a658b4eb0ca758/cd3084b6-b7b5-463e-b334-8c7e6171ed3c)

Error has been fixed.   QA retested Tribal Receipt, Offsets to Operating Expenditures and Other Receipts 
Other Receipts.

QA Review Completed.  Moved to Stage Ready.
